streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java

```diff
@@ -1724,4 +1731,52 @@ private String getSavePointed(Application appParam) {
     }
     return null;
   }
+
+  /** Check queue label validation when create the application if needed. */
+  @VisibleForTesting
+  public void checkQueueValidationIfNeeded(Application appParam) {
+    yarnQueueService.checkQueueLabelFormatIfNeeded(
+        appParam.getExecutionModeEnum(), appParam.getYarnQueue());
+    if (!isYarnPerJobAppModeNotDefaultQueue(appParam)) {
+      return;
+    }
+    boolean exist =
+        yarnQueueService.existByTeamIdQueueLabel(appParam.getTeamId(), appParam.getYarnQueue());
+    ApiAlertException.throwIfFalse(
+        exist, String.format(ERROR_APP_QUEUE_HINT, appParam.getYarnQueue(), appParam.getTeamId()));
+  }
+
+  /** Check queue label validation when update the application if needed. */
+  @VisibleForTesting
+  public void checkQueueValidationIfNeeded(Application oldApp, Application newApp) {
+    yarnQueueService.checkQueueLabelFormatIfNeeded(
+        newApp.getExecutionModeEnum(), newApp.getYarnQueue());
+    if (!isYarnPerJobAppModeNotDefaultQueue(newApp)) {
+      return;
+    }
+
+    oldApp.setYarnQueueByHotParams();
+    if (ExecutionMode.isYarnPerJobOrAppMode(newApp.getExecutionModeEnum())
+        && StringUtils.equals(oldApp.getYarnQueue(), newApp.getYarnQueue())) {
+      return;
+    }
+    boolean exist =
+        yarnQueueService.existByTeamIdQueueLabel(newApp.getTeamId(), newApp.getYarnQueue());
+    ApiAlertException.throwIfFalse(
+        exist, String.format(ERROR_APP_QUEUE_HINT, newApp.getYarnQueue(), newApp.getTeamId()));
+  }
+
+  /**
+   * Judge the execution mode whether is the Yarn PerJob or Application mode with not default or
+   * empty queue label.
+   *
+   * @param application application entity.
+   * @return If the executionMode is (Yarn PerJob or application mode) and the queue label is not
+   *     (empty or default), return true, false else.
+   */
+  @VisibleForTesting
+  public boolean isYarnPerJobAppModeNotDefaultQueue(Application application) {
```

Comment:
We can change `public` to `private` and method rename to `isYarnNotDefaultQueue`